samples/apps/copilot-chat-app/documentuploadapp/Program.cs

```diff
@@ -0,0 +1,160 @@
+        {
+            Console.WriteLine($"Error: {ex.Message}");
+            return null;
+        }
+    }
+
+    /// <summary>
+    /// Conditionally uploads a file to the Document Store for parsing.
+    /// </summary>
+    /// <param name="file">The file to upload for injection.</param>
+    /// <param name="serviceConfig">The service configuration.</param>
+    /// <param name="toUserCollection">Save the extracted context to an isolated user collection.</param>
+    private static async Task UploadFileAsync(FileInfo file, ServiceConfig serviceConfig, bool toUserCollection)
+    {
+        if (!file.Exists)
+        {
+            Console.WriteLine($"File {file.FullName} does not exist.");
+            return;
+        }
+
+        if (toUserCollection)
+        {
+            var userId = await AcquireUserIdAsync(serviceConfig);
+            if (userId != null)
+            {
+                Console.WriteLine("Uploading and parsing file to user collection...");
+                await UploadFileAsync(file, userId, serviceConfig);
+            }
+        }
+        else
+        {
+            Console.WriteLine("Uploading and parsing file to global collection...");
+            await UploadFileAsync(file, "", serviceConfig);
+        }
+    }
+
+    /// <summary>
+    /// Invokes the DocumentQuerySkill to upload a file to the Document Store for parsing.
+    /// </summary>
+    /// <param name="file">The file to upload for injection.</param>
+    /// <param name="userId">The user unique ID. If empty, the file will be injected to a global collection that is available to all users.</param>
+    /// <param name="serviceConfig">The service configuration.</param>
+    private static async Task UploadFileAsync(
+        FileInfo file, string userId, ServiceConfig serviceConfig)
+    {
+        string skillName = "DocumentQuerySkill";
+        string functionName = "ParseLocalFile";
+        string commandPath = $"skills/{skillName}/functions/{functionName}/invoke";
+
+        using StringContent jsonContent = new(
+            JsonSerializer.Serialize(
+                new
+                {
+                    input = file.FullName,
+                    Variables = new List<KeyValuePair<string, string>> {
+                        new KeyValuePair<string, string>("userId", userId)
+                    }
+                }
+            ),
+            Encoding.UTF8,
+            "application/json"
+        );
+
+        // Create a HttpClient instance and set the timeout to infinite since
+        // large documents will take a while to parse.
+        using HttpClientHandler clientHandler = new()
+        {
+            CheckCertificateRevocationList = true
+        };
+        using HttpClient httpClient = new(clientHandler)
+        {
+            Timeout = Timeout.InfiniteTimeSpan
+        };
+
+        try
+        {
+            using HttpResponseMessage response = await httpClient.PostAsync(
```

Comment:
We should support uploading the file through the API. 
https://learn.microsoft.com/en-us/aspnet/core/mvc/models/file-uploads?view=aspnetcore-6.0

Reply:
Future PR